pkg/policies/flowcontrol/actuators/workload-scheduler/scheduler.go

```diff
@@ -425,6 +435,12 @@ func (s *Scheduler) Decide(ctx context.Context, labels labels.Labels) iface.Limi
 			// dropped if it doesn't get the tokens immediately.
 			timeout = 0
 		}
+
+		// find the minimum of matchedWorkloadTimeout and client's timeout
+		if hasWorkloadTimeout && matchedWorkloadTimeout < timeout {
```

Comment:
@hasit,

This doesn't cover the scenario when `clientDeadline` is not defined (0) and workload timeout is defined

Reply:
Also, the behavior might be unexpected in the reverse scenario when `clientDeadline` is defined and workload timeout is not defined

Reply:
Addressed in https://github.com/fluxninja/aperture/pull/2336.